addon/-task-property.js

```diff
@@ -408,6 +408,10 @@ objectAssign(TaskProperty.prototype, propertyModifiers, {
   constructor: TaskProperty,
 
   setup(proto, taskName) {
+    if (this._hasSetMaxConcurrency && !this._hasSetBufferPolicy) {
```

Comment:
Maybe instead of tracking `_hasSetMaxConcurrency` you just replace it with a check of `this._maxConcurrency === Infinity`? The fewer the properties to keep track of the better, and now that I think about it it's probably the most logical thing we're trying to prevent (the idea that when MC is infinity, all the modifiers behave the same). For reference, [this is where the default maxConcurrency is set to Infinity](https://github.com/machty/ember-concurrency/blob/72f70b6c327f5242ca623d61ea0595b5f9093896/addon/-property-modifiers-mixin.js#L13).